sap-service-operator/src/main/java/com/sap/cloud/environment/servicebinding/FileSystemWatcherCache.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+package com.sap.cloud.environment.servicebinding;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.sap.cloud.environment.servicebinding.api.ServiceBinding;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_DELETE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+
+class FileSystemWatcherCache implements DirectoryBasedCache
+{
+    @Nonnull
+    private static final Collection<WatchEvent.Kind<?>> MODIFICATION_EVENTS =
+        Arrays.asList(ENTRY_CREATE, ENTRY_MODIFY, ENTRY_DELETE);
+
+    @Nonnull
+    private final Function<Path, ServiceBinding> serviceBindingLoader;
+    @Nonnull
+    // internal for simplified testing
+    final Map<Path, ServiceBinding> cachedServiceBindings = new HashMap<>();
```

Comment:
(Minor remark)
I was stumbling over "internal" - ignore if it's that an established synonym
```suggestion
    // package-private for simplified testing
    final Map<Path, ServiceBinding> cachedServiceBindings = new HashMap<>();
```